driver/src/orderbook/streamed/order.rs

```diff
@@ -1,19 +1,106 @@
 use super::*;
+use crate::contracts::stablex_auction_element;
+use crate::models::Order as ModelOrder;
+use anyhow::Result;
 use serde::{Deserialize, Serialize};
 
+/// Change from a potential solution that might still get replaced by a better solution in the same
+/// batch.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+struct PendingUsedAmount {
+    batch_id: BatchId,
+    amount: u128,
+}
+
 #[derive(Copy, Clone, Debug, Deserialize, Serialize)]
 pub struct Order {
     pub buy_token: TokenId,
     pub sell_token: TokenId,
-    pub valid_from: BatchId,
     pub valid_until: BatchId,
-    pub price_numerator: u128,
-    pub price_denominator: u128,
-    pub used_amount: u128,
+    pub valid_from: BatchId,
+    price_numerator: u128,
+    price_denominator: u128,
+    // Invariant: used_amount + pending_used_amount.amount <= price_denominator
```

Comment:
I assume we can enforce this invariant as soon as we start writing the pending & used_amount.

Reply:
In this version the used amount can't change so the invariant is enforced :) . It is going to change and be enforced in the `trade` and `trade_reversion` functions that I left unimplemented in this PR.